[docs] Gh #5196 describe current limitations (#5428)
DevExpress/testcafe: docs/articles/documentation/guides/continuous-integration/github-actions.md

This job runs on a GitHub-hosted virtual machine with the latest Windows version. `test` is the [job ID](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_id) that must be unique to the `jobs` object.

> > You can use a GitHub-hosted virtual machine with a variety of operating systems to run tests, as listed on [this page](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on). For simplicity, all examples in this article run on `windows-latest`.

Reviewer: ```suggestion
> You can use a GitHub-hosted virtual machine with a variety of operating systems to run tests, as listed on the following page: [GitHub Docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on). For simplicity, all examples in this article run on `windows-latest`.
```

> Github Actions uses the `macOS Catalina 10.15` virtual environment with *"System Integrity Protection"* enabled as `macos-latest`. With this setting enabled, TestCafe requires a screen recording permission, which is unobtainable programmatically. For this reason, TestCafe is unable to run tests locally on `macos-latest` in GitHub Actions.

Reviewer: ```suggestion
Github Actions uses the `macOS Catalina 10.15` virtual environment with *"System Integrity Protection"* enabled as `macos-latest`. With this setting enabled, TestCafe requires screen recording permission, which cannot be obtained programmatically. For this reason, TestCafe is unable to run tests locally on `macos-latest` in GitHub Actions.
```

> However, tests do run in the remote mode.

Reviewer 2: 'Remote mode' looks like a term but we do not have such a term. I guess it's better to say:

*However, tests can run on macOS VMs if you connect the browser as remote.*

Author: edited

> However, tests can run on macOS VMs if you connect the browser as remote.

Reviewer: Is "VMs" short for "virtual machines"? If so, I think you should spell it out.

Author: That's right, thanks

Reviewer: Thanks, please apply the change.

Author: done

**Example**

```sh
export HOSTNAME=localhost
export PORT1=1337
export PORT2=1338
testcafe remote test.js --hostname ${HOSTNAME} --ports ${PORT1},${PORT2} &
pid=$!
open -a Safari http://${HOSTNAME}:${PORT1}/browser/connect
wait $pid
```

Alternatively, you can [**host your own runners**](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) for the job. This gives you more precise control over the environment.

To set up the self-hosted runners, [add them to your repository](https://docs.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners#adding-a-self-hosted-runner-to-a-repository).

> Afterwards, configure `runs-on` in your workflow `.yml` file:

Reviewer 2: `Afterwards` is usually not recommended in the technical context, it's too figurative, and it also implies 'at a later time', 'after some time passes'.

`After that` is better as a more neutral alternative.

Author: edited
